File: C++/Homework_15/Homework_15/Circle.cpp
#include "Circle.h"

bool Circle::operator==(const Circle& other) const
{
	return this->radius == other.radius;
}
bool Circle::operator>(const Circle& other) const
{
	return this->radius > other.radius;
}
bool Circle::operator<(const Circle& other) const
{
	return this->radius < other.radius;
}
Circle& Circle::operator+=(double value) {
	radius += value;
	return *this;
}
Circle& Circle::operator-=(double value) {
	radius -= value;
	return *this;
}


double Circle::get() const {
	return this->radius;
}
void Circle::print(string text) const {
	cout << text << " " << this->radius << endl;
	
}

File: C#/Homework_08/Homework_08/Circle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Homework_08
{
    class Circle : GeometricFigure
    {
        private double radius;
        public Circle(double radius)
        {
            this.radius = radius;
        }

        public override double GetArea()
        {
            return Math.PI * radius * radius;
        }

        public override double GetPerimeter()
        {
            return 2 * Math.PI * radius;
        }
    }
}
